tests/providers/ptq_test.py

```diff
@@ -155,51 +157,63 @@ def test_nnx_ptq(self):
         out_features=5,
         rngs=nnx.Rngs(0),
         kernel_init=nnx.with_partitioning(
-            nnx.initializers.lecun_normal(), ("contraction", "remaining")
+            nnx.initializers.lecun_normal(), ("contraction", None)
         ),
     )
+
+    unsharded_state = nnx.state(fp_linear)
+    sharding = nnx.get_named_sharding(unsharded_state, mesh)
+    sharded_state = jax.device_put(unsharded_state, sharding)
+    nnx.update(fp_linear, sharded_state)
     # Weight quantization method 1: use quantize_model to convert both the
     # model and params, i.e., implicit quantization.
-    ptq_linear = qwix_model.quantize_model(
-        fp_linear,
-        ptq.PtqProvider(q_rules),
-        model_input,
-    )
-    # Test PTQ param structure.
-    qw = ptq_linear.kernel
-    self.assertIsInstance(qw, ptq.WithAux)
-    self.assertEqual(qw.weight_name, "kernel")
-    qw = qw.array
-    self.assertEqual(qw.qvalue.dtype, jnp.int8)
-    self.assertEqual(qw.qvalue.shape, (12, 5))
-    self.assertEqual(qw.qvalue.sharding, ("contraction", "remaining"))
-    self.assertEqual(qw.scale.shape, (3, 5))
-    self.assertEqual(qw.scale.sharding, ("contraction", "remaining"))
-
-    # Weight quantization method 2: call quantize_model in eval_shape and
-    # quantize_params.
-    abs_ptq_linear = nnx.eval_shape(
-        lambda: qwix_model.quantize_model(
-            fp_linear,
-            ptq.PtqProvider(q_rules),
-            model_input,
-        ),
-    )
-    # Test manual quantize_params.
-    orig_params = nnx.state(fp_linear, nnx.Param)
-    orig_params = nnx.to_pure_dict(orig_params)
-    quantized_params = ptq.quantize_params(orig_params, abs_ptq_linear)
-    # quantized_params can be updated to abs_ptq_linear.
-    nnx.update(abs_ptq_linear, quantized_params)
-    # Ensure that the model can be called.
-    abs_ptq_linear(model_input)
+    with mesh:
+      ptq_linear = qwix_model.quantize_model(
+          fp_linear,
+          ptq.PtqProvider(q_rules),
+          model_input,
+      )
+      # Test PTQ param structure.
+      qw = ptq_linear.kernel
+      self.assertIsInstance(qw, ptq.WithAux)
+      self.assertEqual(qw.weight_name, "kernel")
+      qw = qw.array
+      self.assertEqual(qw.qvalue.dtype, jnp.int8)
+      self.assertEqual(qw.qvalue.shape, (12, 5))
+      self.assertEqual(
+          qw.qvalue.sharding,
+          jax.sharding.NamedSharding(mesh, P()),
```

Comment:
To make this correct you also need to shard the model_input to (None, "contraction") before calling quantize_model.